pkg/infinity/httptrace.go

```diff
@@ -0,0 +1,53 @@
+package infinity
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"net/http/httptrace"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+func ApplyHTTPTraceToRequest(_ context.Context, req *http.Request) *http.Request {
+	ctx := req.Context()
+	logger := backend.Logger.FromContext(ctx)
+	trace := &httptrace.ClientTrace{
+		GetConn: func(hostPort string) {
+			logger.Debug("HTTPTrace event", "event_name", "GetConn")
```

Comment:
I am wondering if all of these should be level `trace` because for data source services, we have `debug` level logging turned on. This would produce a lot of logs that are only sometimes needed. So we could turn it on in that case. 

We have `trace` log level: https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/log/log.go#L14, but we are missing `logger.Trace()` method - all other levels have it.

Reply:
I'm with Ivana here. Why not just even be a trace instead of log?

Reply:
I was thinking about a proper trace so these would be spans instead of logs. Like in here https://github.com/AdamKorcz/skipper/blob/d0b249a4aa8993a050b87cecc11cc6ca839195a4/proxy/proxy.go#L1464